conventions/src/main/kotlin/io.opentelemetry.instrumentation.base.gradle.kts

```diff
@@ -119,3 +121,35 @@ if (testLatestDeps) {
     }
   }
 }
+
+val otelInstrumentation = extensions.create<OtelInstrumentationExtension>("otelInstrumentation")
+
+tasks {
+  val generateInstrumentationVersionFile by registering {
+    val propertiesDir = File(project.buildDir, "generated/instrumentationVersion")
+    outputs.dir(propertiesDir)
```

Comment:
We need to add version as `inputs`. Also consider registering the subdirectory to outputs instead of this top level one, which causes directories to be automatically created. So maybe start by copying from SDK repo :)

https://github.com/open-telemetry/opentelemetry-java/blob/main/buildSrc/src/main/kotlin/otel.java-conventions.gradle.kts#L143

Reply:
Nice, thanks!